security/pkg/nodeagent/plugin/providers/google/iamclient.go

```diff
@@ -50,5 +63,30 @@ func NewPlugin() iam.IAMCredentialsClient {
 		return nil
 	}
 
-	return iam.NewIAMCredentialsClient(conn)
+	return Plugin{
+		iamClient: iam.NewIAMCredentialsClient(conn),
+	}
+}
+
+// ExchangeToken exchange oauth access token from trusted domain and k8s sa jwt.
+func (p Plugin) ExchangeToken(ctx context.Context, trustedDomain, k8sSAjwt string) (string /*access token*/, time.Time /*expireTime*/, error) {
+	req := &iam.GenerateIdentityBindingAccessTokenRequest{
+		Name:  fmt.Sprintf("projects/-/serviceAccounts/%s", trustedDomain),
```

Comment:
Why we have a '-' behind projects?

Reply:
added comments, it's required by the api, similar to [generateAccessToken](https://cloud.google.com/iam/credentials/reference/rest/v1/projects.serviceAccounts/generateAccessToken): resource name of the service account for which the credentials are requested, in the following format: projects/-/serviceAccounts/{ACCOUNT_EMAIL_OR_UNIQUEID}

Reply:
Okay, thanks for the clarification.